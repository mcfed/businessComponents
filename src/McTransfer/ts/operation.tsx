import React from 'react';
import {Button, Icon} from 'antd';
import {ButtonType, ButtonProps} from 'antd/lib/button';

export function noop() {}

/**
 * moveToLeft：数据移向左侧事件，必须
 * moveToRight：数据移向右侧事件，必须
 * leftArrowText：左箭头text，非必须
 * rightArrowText：右箭头text，非必须
 * leftActive：左箭头是否disabled，非必须
 * rightActive：右箭头是否disabled，非必须
 * className：整个操作模块className 前缀，非必须
 */

interface ItemProps {
  moveToLeft: () => void;
  moveToRight: () => void;
  leftArrowText?: string;
  rightArrowText?: string;
  leftActive?: boolean;
  rightActive?: boolean;
  className?: string;
}

export const Operation: React.FC<ItemProps> = (props: ItemProps) => {
  const {
    leftActive,
    rightActive,
    moveToLeft,
    moveToRight,
    leftArrowText,
    rightArrowText,
    className
  } = props;
  const moveToLeftButton = (
    <Button
      // v3.9 bug,编译报错
      type={'primary' as any}
      size='small'
      //@ts-ignore
      disabled={!leftActive}
      onClick={moveToLeft}>
      {
        <span>
          <Icon type='left' />
          {leftArrowText}
        </span>
      }
    </Button>
  );
  const moveToRightButton = (
    <Button
      // v3.9 bug,编译报错
      type={'primary' as any}
      size='small'
      //@ts-ignore
      disabled={!rightActive}
      onClick={moveToRight}>
      {
        <span>
          {rightArrowText}
          <Icon type='right' />
        </span>
      }
    </Button>
  );
  return (
    <div className={className}>
      {moveToRightButton}
      {moveToLeftButton}
    </div>
  );
};
