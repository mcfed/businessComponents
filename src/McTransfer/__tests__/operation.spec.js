import React from 'react';
import {shallow} from 'enzyme';
import Operation from '../operation';
import * as Op from '../ts/operation.tsx';

function setup(props, options) {
  const defaultProps = {
    moveToLeft: jest.fn(),
    moveToRight: jest.fn(),
    leftArrowText: '',
    rightArrowText: '',
    leftActive: true,
    rightActive: true,
    className: ''
  };
  const wrapper = shallow(
    <Operation {...Object.assign({}, defaultProps, props)} />,
    options
  );

  return {
    wrapper
  };
}

describe('快照测试', () => {
  it('全页快照', () => {
    const props = {
      item: {}
    };
    const {wrapper} = setup(props);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('方法测试全覆盖', () => {
  it('noop 方法测试', () => {
    Op.noop();
  });
});
