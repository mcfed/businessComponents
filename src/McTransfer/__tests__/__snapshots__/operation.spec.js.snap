// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 全页快照 1`] = `
<div
  className=""
>
  <Button
    block={false}
    disabled={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
    size="small"
    type="primary"
  >
    <span>
      <Icon
        type="right"
      />
    </span>
  </Button>
  <Button
    block={false}
    disabled={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[MockFunction]}
    size="small"
    type="primary"
  >
    <span>
      <Icon
        type="left"
      />
    </span>
  </Button>
</div>
`;
